### DISCUSSION TITLE
PWA solution

### DISCUSSION THREAD

**Discussion by ehsan-g** [OutlineFoundation/outline-sdk#81] [Q&A] [answered]
I was wondering if utilising the Side Service method works for PWA  applications?

**Comment by fortuna** [comment, marked as answer]
Progressive Web Apps don't have control of the network stack, so they can't really leverage the SDK.
However, you may be able to package a web app in an native app, which in turn can be integrated with the SDK. 
See https://github.com/Jigsaw-Code/outline-sdk/tree/main/x/examples/outline-connectivity-app for an example.
![image](https://github.com/Jigsaw-Code/outline-sdk/assets/113565/afed6efb-2be7-4331-bec3-099706dec86d)
